cmip6_downscaling/methods/common/tasks.py

```diff
@@ -55,29 +55,33 @@ def get_obs(run_parameters: RunParameters) -> UPath:
 
 
 @task
-def get_experiment(run_parameters: RunParameters) -> UPath:
-
-    # TODO: get train and predict data here
+def get_experiment(run_parameters: RunParameters, time_subset: str) -> UPath:
 
     target = (
         intermediate_dir
         / "get_experiment"
+        / time_subset
         / "{model}_{scenario}_{variable}_{latmin}_{latmax}_{lonmin}_{lonmax}_{train_dates[0]}_{train_dates[1]}_{predict_dates[0]}_{predict_dates[1]}".format(
             **asdict(run_parameters)
         )
     )
     if use_cache and (target / '.zmetadata').exists():
         print(f'found existing target: {target}')
         return target
-
+    print(target)
     ds = load_cmip(
         source_ids=run_parameters.model, return_type='xr', variable_ids=run_parameters.variable
     ).pipe(lon_to_180)
 
-    subset = subset_dataset(
-        ds, run_parameters.variable, run_parameters.train_period.time_slice, run_parameters.bbox
-    )
-    subset = subset.chunk({'time': 365})  # TODO: do better here
+    if time_subset == 'train_period':
+        time_period = run_parameters.train_period.time_slice
+    elif time_subset == 'predict_period':
+        time_period = run_parameters.predict_period.time_slice
+    else:
+        raise ValueError('time_subset arg is not train_period or predict_period.')
```

Comment:
```suggestion
time_period = getattr(run_parameters, time_subset)
```

I think will do the same thing.

Reply:
Ah this is way cleaner!